drivers/include/net/netdev/ieee802154.h

```diff
@@ -128,7 +135,12 @@ typedef struct {
 /**
  * @brief   Received packet status information for IEEE 802.15.4 radios
  */
-typedef struct netdev_radio_rx_info netdev_ieee802154_rx_info_t;
+typedef struct netdev_ieee802154_rx_info {
+    uint64_t timestamp;     /**< Timestamp value of a received frame in ns */
+    int16_t rssi;           /**< RSSI of a received frame in dBm */
+    uint8_t lqi;            /**< LQI of a received frame */
+    uint8_t flags;          /**< Flags e.g. used to mark other fields as valid */
```

Comment:
I don't like adding those 8 bytes for everyone unconditionally

How about

```suggestion
    int16_t rssi;           /**< RSSI of a received frame in dBm */
    uint8_t lqi;            /**< LQI of a received frame */
    uint8_t flags;          /**< Flags e.g. used to mark other fields as valid */
#if IS_USED(MODULE_GNRC_NETIF_TIMESTAMP)
    uint64_t timestamp;     /**< Timestamp value of a received frame in ns */
#endif
```

`flags` is 'free' due to padding and can be used for other additions as well, I like this as it's useful for #16312 too